### PR TITLE
Task as/j114 add markup for edit podcast page

### PR DESCRIPTION
**Pull request by qwiz1**

### CODE REVIEW

File: frontend/src/components/common/common.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export { default as Loader } from './loader/loader';
 export { default as Select } from './select/select';
 export { default as Datepiker } from './datepicker/datepicker';
 export { default as Table } from './table/table';
+export { default as InputImage } from './input-image/input-image';
```
Comment (ghost) [inline frontend/src/components/common/common.ts]:
```suggestion
export { default as ImagePreviewControl } from './image-preview-control/image-preview-control';
```

File: frontend/src/components/common/input-image/input-image.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      <div className={styles.panel}>
+        <img
+          src={cameraIcon}
+          width="24"
+          height="20"
+          loading="lazy"
+          alt=""
+          className={styles.cameraIcon}
+        />
+      </div>
```
Comment (ghost) [inline frontend/src/components/common/input-image/input-image.tsx]:
```suggestion
```
do it as `::before` in css

File: frontend/src/components/common/input-image/input-image.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+const acceptExtension = getFileExtensions(
+  FileExtension.JPEG,
+  FileExtension.JPG,
+  FileExtension.PNG,
+  FileExtension.SVG,
+);
+
+const InputImage: React.FC<Props> = ({ register, image }) => {
```
Comment (ghost) [inline frontend/src/components/common/input-image/input-image.tsx]:
```suggestion
const ImagePreviewControl: React.FC<Props> = ({ register, image }) => {
```

File: frontend/src/components/configurate-podcast/styles.module.scss [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
   display: flex;
   flex-direction: column;
-  max-width: 1515px;
+  max-width: 716px;
   width: 100%;
   margin: 0 auto;
-  padding: 30px;
+  padding: 0;
+
+  h2{
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/styles.module.scss]: add `class` for it

File: frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx [hunk 5]
```diff
@@ -37,29 +30,26 @@ const ConfiguratePodcastForm: React.FC<Props> = ({
     defaultValues: payload,
   });
 
-  const { createPodcastStatus } = useAppSelector(({ configuratePodcast }) => ({
+  const { createPodcastStatus, image } = useAppSelector(({ configuratePodcast }) => ({
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx]:
```suggestion
  const { createPodcastStatus } = useAppSelector(({ configuratePodcast }) => ({
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx [hunk 6]
```diff
@@ -37,29 +30,26 @@ const ConfiguratePodcastForm: React.FC<Props> = ({
...
     createPodcastStatus: configuratePodcast.dataStatus,
+    image: configuratePodcast.podcast?.image,
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx]:
```suggestion
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx [hunk 7]
```diff
@@ -37,29 +30,26 @@ const ConfiguratePodcastForm: React.FC<Props> = ({
...
   }));
 
   const isFormDisabled = createPodcastStatus === DataStatus.PENDING;
 
   return (
     <form className={styles.form} onSubmit={handleSubmit(onSubmit)}>
       <fieldset disabled={isFormDisabled} className={styles.fieldset}>
+        <InputImage
+          register={register}
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx]: try to do it with `useController`

File: frontend/src/components/common/select/select.tsx [hunk 8]
```diff
@@ -20,6 +20,24 @@ type Props = {
   errors: FieldErrors;
 };
 
+type OptionType = {
+  label: string;
+  value: string;
+};
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]:
```suggestion
```
we already have it

File: frontend/src/components/common/select/select.tsx [hunk 9]
```diff
@@ -20,6 +20,24 @@ type Props = {
...
+
+const customStyles: StylesConfig<OptionType, false> = {
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]: move to `frontend/src/components/common/select/styles.ts`

Comment (ghost) [inline frontend/src/components/common/select/select.tsx]:
```suggestion
const styles: StylesConfig<OptionType, false> = {
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/image-preview-control/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  max-width: 716px;
+  width: 100%;
+  cursor: pointer;
+
+  &::before{
+    content: '';
+    position: absolute;
+    bottom: 0;
+    width: 100%;
+    height: 54px;
```
Comment (ghost) [inline frontend/src/components/common/image-preview-control/styles.module.scss:13]:
```suggestion
    height: 100%;
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx [hunk 11]
```diff
@@ -47,20 +39,17 @@ const ConfiguratePodcastForm: React.FC<Props> = ({
   return (
     <form className={styles.form} onSubmit={handleSubmit(onSubmit)}>
       <fieldset disabled={isFormDisabled} className={styles.fieldset}>
-        <Input
-          name={PodcastPayloadKey.NAME}
+        <ImagePreviewControl
+          name={PodcastPayloadKey.IMAGE}
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx:44]: hmm, do we need to pass an existing image? 🤔

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx (hunk 11 above)
Comment (qwiz1) [reply frontend/src/components/configurate-podcast/components/configurate-podcast-form/configurate-podcast-form.tsx:44]: > hmm, do we need to pass an existing image? 🤔

I think so.
If I don't need it, may you tell me why?

File: frontend/src/components/common/image-preview-control/styles.module.scss (hunk 10 above)
Comment (qwiz1) [reply frontend/src/components/common/image-preview-control/styles.module.scss:13]: If you apply this style, the mask will be on the whole picture, is that how it should be?
image

File: frontend/src/components/common/image-preview-control/image-preview-control.tsx [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+const ImagePreviewControl: React.FC<Props> = ({ name, control, errors, imageUrl }) => {
+  const { field } = useController({ name, control });
+
+  const handleChange = (evt: React.ChangeEvent<FieldValues>): void => {
+    const hasImg = Boolean(evt.target.files.length);
+    const imgUrl = hasImg
+      ? URL.createObjectURL(evt.target.files[0])
+      : imageUrl ?? defaultImage;
+
+    field.onChange(imgUrl);
```
Comment (ghost) [inline frontend/src/components/common/image-preview-control/image-preview-control.tsx]:
```suggestion
    const [file] = evt.target.files ?? [];
    const imgUrl = hasImg
      ? URL.createObjectURL(file)
      : imageUrl ?? defaultImage;

    field.onChange(imgUrl);
```

File: frontend/src/components/common/image-preview-control/image-preview-control.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    <>
+      <label className={styles.inputWrapper}>
+        <img
+          src={field.value ?? imageUrl ?? defaultImage}
+          width="716"
+          height="281"
+          loading="lazy" alt=""
+        />
+        <input
+          {...control.register(name)}
```
Comment (ghost) [inline frontend/src/components/common/image-preview-control/image-preview-control.tsx:54]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**